File: OpenSilver.Samples.TelerikUI/OpenSilver.Samples.TelerikUI/Samples/Controls/RadChart3D/ViewModel/RadialViewModel.vb
Imports OpenSilver.Samples.TelerikUI.RadChart3D.Helpers
Imports System.Collections.Generic

Namespace OpenSilver.Samples.TelerikUI.RadChart3D.ViewModel
    Public NotInheritable Class RadialViewModel
        Private _data As IEnumerable(Of Double)

        Public ReadOnly Property Data As IEnumerable(Of Double)
            Get
                If _data Is Nothing Then
                    _data = RadChart3D.Helpers.SeriesExtensions.GetUserRadialData()
                End If

                Return _data
            End Get
        End Property
    End Class
End Namespace

File: OpenSilver.Samples.TelerikUI/OpenSilver.Samples.TelerikUI/Samples/Controls/RadChart3D/ViewModel/RadialViewModel.cs
using OpenSilver.Samples.TelerikUI.RadChart3D.Helpers;
using System;
using System.Collections.Generic;
using System.Text;

namespace OpenSilver.Samples.TelerikUI.RadChart3D.ViewModel
{
    public sealed class RadialViewModel
    {
        private IEnumerable<double> _data;

        public IEnumerable<double> Data
        {
            get
            {
                if (_data == null)
                {
                    _data = SeriesExtensions.GetUserRadialData();
                }

                return _data;
            }
        }
    }
}
